datafusion-cli/src/catalog.rs

```diff
@@ -178,13 +180,18 @@ impl SchemaProvider for DynamicFileSchemaProvider {
                 // to any command options so the only choice is to use an empty collection
                 match scheme {
                     "s3" | "oss" | "cos" => {
-                        state = state.add_table_options_extension(AwsOptions::default());
+                        if let Some(table_options) = builder.table_options() {
```

Comment:
It seems like this code previously always registered the table options and now it only registers them if another table option was registered. Is that intentional? 

Maybe it doesn't matter

Reply:
Well .... I updated all of the existing code in the project to reflect the change but I agree that it may be a bit of a surprise to anyone expecting the old behavior when upgrading. 6 of one, 1/2 dozen of another ...

Reply:
Wait, sorry, misread the comment. I think it works only because the builder was created from an existing session state.

Reply:
Maybe we can clean it up as a follow on -- it is not critical in my opinion